lib/utils/compose_ts.ts

```diff
@@ -1405,6 +1418,18 @@ export function createRunLoop(tick: (...args: any[]) => void) {
 	return runloop;
 }
 
+async function tryGetContractContent(
```

Comment:
@toochevere, your commit looks like a great contribution! 👍 
Does it go (or would it go) as far as addressing issue #2183 [balena deploy should parse, validate, and create a contract as part of the release it creates](https://github.com/balena-io/balena-cli/issues/2183) ?
If your commit does not depend on Felipe's commit, and even more so if it addresses #2183, I would suggest creating a separate PR for it.

Reply:
Not sure entirely about #2183, but I think PJ's commit will allow to close #1908, since this will add the version from the contract to the release during deploy